aten/src/ATen/native/cuda/CuFFTPlanCache.h

```diff
@@ -420,37 +419,30 @@ class CuFFTParamsLRUCache {
 
   void resize(int64_t new_size) {
     _set_max_size(new_size);
-    // no-op when compiling with CUDA 10.
+    // no-op when compiling with CUDA 10 since we don't have to
+    // do the workaround with the 1024th cufft plan bug
     #if CUDA_VERSION < 10000
-    auto cur_size = _usage_list.size();
-    if (cur_size > _max_size) {
-      auto delete_it = _usage_list.end();
-      for (size_t i = 0; i < cur_size - _max_size; i++) {
-        delete_it--;
-        _cache_map.erase(delete_it->first);
+      auto cur_size = _usage_list.size();
```

Comment:
So these are not a work around for the 1024th plan? I am not sure what this block of code does but I interpreted it as since the last element from the cache is removed, may be it's removing 1024th plan?

Reply:
You are right. These are not. The cache size limit is here because cufft plan may use GPU memory, and GPU resource is valuable.

The default max_size `1024` was determined by the cufft bug.

Reply:
This blocks of code frees extra plans if the new max size is smaller than the number of plans in the cache.